internal/allocator/allocator_test.go

```diff
@@ -316,36 +318,50 @@ func TestAssignment(t *testing.T) {
 		{
 			desc: "s4 takes s3's former IP",
 			svc:  "s4",
-			ip:   "1000::4:ff",
+			ips:  []string{"1000::4:ff"},
+		},
+		// IP dual-stack test
+		{
+			desc: "s2 frees its IP",
+			svc:  "s2",
+			ips:  []string{},
+		},
+		{
+			desc: "ip dual-stack assign s1",
+			svc:  "s1",
+			ips:  []string{"1.2.3.4", "1000::4"},
```

Comment:
can we add a couple of tests where we try to assign to ips, one matching one pool and the other matching the other to see if it fails?